applications/main/nfc/plugins/supported_cards/ndef.c

```diff
@@ -40,6 +40,18 @@ static void
     furi_string_cat(str, "\n");
 }
 
+static char decode_char(const char* str) {
+    int high = (str[1] >= 'A' && str[1] <= 'F') ? (str[1] - 'A' + 10) :
+               (str[1] >= 'a' && str[1] <= 'f') ? (str[1] - 'a' + 10) :
+               (str[1] >= '0' && str[1] <= '9') ? (str[1] - '0') :
+                                                  0;
+    int low = (str[2] >= 'A' && str[2] <= 'F') ? (str[2] - 'A' + 10) :
+              (str[2] >= 'a' && str[2] <= 'f') ? (str[2] - 'a' + 10) :
+              (str[2] >= '0' && str[2] <= '9') ? (str[2] - '0') :
+                                                 0;
+    return (high << 4) | low;
+}
```

Comment:
i think there is a function like this in flipper libs, ill check

Reply:
there would be sscanf() which would be cleaner code, but for this usecase it is overkill, uses much more stack

Reply:
would this work:

```
static inline int hex_to_int(char c) {
    if (c >= '0' && c <= '9') return c - '0';
    if (c >= 'A' && c <= 'F') return c - 'A' + 10;
    if (c >= 'a' && c <= 'f') return c - 'a' + 10;
    return 0;
}

static char decode_char(const char* str) {
    return (hex_to_int(str[1]) << 4) | hex_to_int(str[2]);
}
```

Reply:
i missed this oops. yeah that is a lot clearer code, nice!